src/search/services/unified_search_service.py

```diff
@@ -305,26 +349,83 @@ def _build_post_fields(self, hit) -> dict[str, Any]:
             "renderable_text": getattr(hit, "renderable_text", None),
         }
 
+    def _extract_journal_from_hubs(self, hit) -> str | None:
+        """Extract journal name from hubs where namespace='journal'."""
+        try:
+            hubs = getattr(hit, "hubs", None)
+            if not hubs:
+                return None
+
+            if not isinstance(hubs, (list, tuple)):
+                return None
+
+            for hub in hubs:
+                if not isinstance(hub, dict):
```

Comment:
If a hub is defined would it ever be not a dict?

Generally, if types are wrong I'd rather the application fail with a warning